vars/packageBuildingPipelineDAOSTest.groovy

```diff
@@ -795,7 +795,12 @@ void call(Map pipeline_args) {
                                                    string(name: 'CI_PR_REPOS',
                                                           value: env.JOB_NAME.split('/')[1] + '@' +
                                                                  "${env.BRANCH_NAME}:${env.BUILD_ID}" +
-                                                                 ' ' + cachedCommitPragma('PR-repos'))
+                                                                 ' ' + cachedCommitPragma('PR-repos')),
+                                                    string(name: 'CI_RPM_Ubuntu20_NOBUILD',
+                                                           value: pipeline_args.get('skip-build', true) ||
+                                                                  pipeline_args.get('skip-build-ubuntu-rpm', true) ||
```

Comment:
I'm on the fence about this one.  It's entirely achievable by not having `ubuntu20.04` in the `distros` list but if that is in the `distros` list _and_ `skip-build` is `false` why would we want to skip it?

That said, we should handle all of the distros' `NOBUILD`s here, not just Ubuntu's.

Reply:
Would be better to ignore `skip-build` and only be sensitive to specific exclusions?
```
string(name: 'CI_RPM_Ubuntu20_NOBUILD',
value: pipeline_args.get('skip-build-ubuntu-rpm', true) ||
'ubuntu20.04' !in distros )
```
or
```
string(name: 'CI_RPM_Ubuntu20_NOBUILD',
value: pipeline_args.get('skip-build-el8-rpm', true) ||
'el8' !in distros )
```

Reply:
No.  The purpose of `skip-build` is to indicate whether the dependency has any relevance to the building of DAOS.  Most (in fact all except `go-build-toolchain` at the moment) really don't have any impact on the build of DAOS which is why we default it to `true` but allow some dependencies to override thta.  But given that `go-build-toolchain` is a _build_ tool it does and it should always be tested on every supported distro (Ubuntu not being one quite yet, hence the need for this specific distro skip).

That said, we should be complete in the implementation of this to handle any future per-distro exceptions without having to come back and revisit this code.

Reply:
Do I understand correctly that in the case of `skip-build == true` (the default case), we should set all `..._NOBUILD` to  `true`?

Reply:
That is correct.  The `in distros` condition is only valid/considered when `skip-build == false`.

Reply:
Like this?
```
string(name: 'CI_RPM_Ubuntu20_NOBUILD',
        value: (pipeline_args.get('skip-build', true) ||
                'ubuntu20.04' !in distros ) ||
                pipeline_args.get('skip-build-ubuntu-rpm', false))
```

Reply:
This is really no different than the existing code in this PR:

```
                                                    string(name: 'CI_RPM_Ubuntu20_NOBUILD',
                                                           value: pipeline_args.get('skip-build', true) ||
                                                                  pipeline_args.get('skip-build-ubuntu-rpm', true) ||
                                                                  'ubuntu20.04' !in distros
                                                                  )
```

The additional bracketing doesn't change the overall logic of the 3 `||`'d values.

My points previously were that I am not sure we want to actually have a `skip-build-*-rpm` for any distro and that we need to handle every distros' `CI_RPM_*_NOBUILD` parameter here with corresponding logic for each distro.

So:
1. Remove the `pipeline_args.get('skip-build-ubuntu-rpm', true) ` condition and
2. Duplicate the parameter for each of the `*_NOBUILD` parameters in https://github.com/daos-stack/daos/blob/7ab7e889849777f8760a7898b9ca234ae202fa00/Jenkinsfile#L229-L238.